Rework how data is downloaded in `ReadOnlyProject.fetch*_df()` methods (#63)
neptune-ai/neptune-fetcher: README.md

| `progress_bar`             | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used.                                         |
| `query`                    | `str`, optional                               | `None`              | NQL query string. Example: `"(accuracy: float > 0.88) AND (loss: float < 0.2)"`. Exclusive with the `with_ids`, `custom_ids`, `states`, `owners`, and `tags` parameters. For syntax, see [Neptune Query Language](https://docs.neptune.ai/usage/nql/) in Neptune docs.                                                         |                                                                                                                  |

> | `match_columns_to_filters` | `bool`, optional                              | `False`             | DEPRECATED: The argument is now assumed to be always `True` and will be removed in a future release.<br/><br/>If `True`, the columns regex will only match columns that are present in the runs that pass the run filters. The run filters must match up to 5000 entries. If `False`, the columns regex will match all columns in the project.                                                                                               |

Reviewer: Same as for the other method

```suggestion
| `match_columns_to_filters` | `bool`, optional                              | `True`             | DEPRECATED: The argument is assumed to be `True` and will be removed in a future release.<br/><br/>If `True`, the columns regex only matches against fields that are present in the runs that pass the run filters. The run filters must match no more than 5000 entries. If `False`, the columns regex matches against all fields in the project.                                                                                                                                           |
```

| `progress_bar`             | `bool`, `Type[ProgressBarCallback]`, optional | `None`              | Set to `False `to disable the download progress bar, or pass a type of ProgressBarCallback to [use your own progress bar](https://docs.neptune.ai/usage/querying_metadata/#using-a-custom-progress-bar). If set to `None` or `True`, the default tqdm-based progress bar will be used.                                                                                     |
| `query`                    | `str`, optional                               | `None`              | NQL query string. Example: `"(accuracy: float > 0.88) AND (loss: float < 0.2)"`. Exclusive with the `with_ids`, `custom_ids`, `states`, `owners`, and `tags` parameters. For syntax, see [Neptune Query Language](https://docs.neptune.ai/usage/nql/) in Neptune docs.                                                                                                     |                                                                                                                  |

> | `match_columns_to_filters` | `bool`, optional                              | `False`             | DEPRECATED: The argument is now assumed to be always `True` and will be removed in a future release.<br/><br/>If `True`, the columns regex will only match columns that are present in the runs that pass the run filters. The run filters must match up to 5000 entries. If `False`, the columns regex will match all columns in the project.                                                                                                                                           |

Reviewer: ```suggestion
| `match_columns_to_filters` | `bool`, optional                              | `True`             | DEPRECATED: The argument is assumed to be `True` and will be removed in a future release.<br/><br/>If `True`, the columns regex only matches against fields that are present in the runs that pass the run filters. The run filters must match no more than 5000 entries. If `False`, the columns regex matches against all fields in the project.                                                                                                                                           |
```
